aptos-move/framework/aptos-framework/sources/permissioned_signer.move

```diff
@@ -0,0 +1,629 @@
+            expiration_time: u64
+        }
+    }
+
+    /// The actual permission configuration stored on-chain.
+    ///
+    /// The address that holds `PermissionStorage` will be generated freshly every time a permission
+    /// handle gets created.
+    enum PermissionStorage has key {
+        V1 {
+            /// A hetherogenous map from `Permission` structs defined by each different modules to
+            /// its permission capacity.
+            perms: BigOrderedMap<Any, StoredPermission>
+        }
+    }
+
+    /// Types of permission capacity stored on chain.
+    enum StoredPermission has store, copy, drop {
+        /// Unlimited capacity.
+        Unlimited,
+        /// Fixed capacity, will be deducted when permission is used.
+        Capacity(u256),
+    }
+
+    /// Create an ephermeral permission handle based on the master signer.
+    ///
+    /// This handle can be used to derive a signer that can be used in the context of
+    /// the current transaction.
+    public fun create_permissioned_handle(master: &signer): PermissionedHandle {
+        assert_master_signer(master);
+        let permissions_storage_addr = generate_auid_address();
+        let master_account_addr = signer::address_of(master);
+
+        move_to(
+            &create_signer(permissions_storage_addr),
+            // Each key is ~100bytes, the value is 12 bytes.
+            PermissionStorage::V1 { perms: big_ordered_map::new_with_config(40, 35, false, 0) }
+        );
+
+        PermissionedHandle::V1 { master_account_addr, permissions_storage_addr }
+    }
+
+    /// Create an storable permission handle based on the master signer.
+    ///
+    /// This handle can be used to derive a signer that can be stored by a smart contract.
+    /// This is as dangerous as key delegation, thus it remains public(package) for now.
+    ///
+    /// The caller should check if `expiration_time` is not too far in the future.
+    public(package) fun create_storable_permissioned_handle(
+        master: &signer, expiration_time: u64
+    ): StorablePermissionedHandle acquires GrantedPermissionHandles {
+        assert_master_signer(master);
+        let permissions_storage_addr = generate_auid_address();
+        let master_account_addr = signer::address_of(master);
+
+        assert!(
+            timestamp::now_seconds() < expiration_time,
+            error::permission_denied(E_PERMISSION_EXPIRED)
+        );
+
+        if (!exists<GrantedPermissionHandles>(master_account_addr)) {
+            move_to<GrantedPermissionHandles>(
+                master, GrantedPermissionHandles { active_handles: vector::empty() }
+            );
+        };
+
+        GrantedPermissionHandles[master_account_addr]
+            .active_handles.push_back(permissions_storage_addr);
+
+        move_to(
+            &create_signer(permissions_storage_addr),
+            PermissionStorage::V1 { perms: big_ordered_map::new_with_config(40, 35, false, 0) }
```

Comment:
since you have this in two places, can you create a function that creates empty PermissionStorage? (i.e. `PermissionStorage::V1 { perms: big_ordered_map::new_with_config(40, 35, false, 0) }` )